doc/source/tutorial/stats/sampling_pinv.rst

```diff
@@ -0,0 +1,271 @@
+    >>> rng_hermite.intervals
+    3340
+    >>> rng_poly = NumericalInversePolynomial(norm(), u_resolution=1e-12)
+    >>> rng_poly.n_intervals
+    252
+
+When exact CDF of a distribution is available, one can estimate the u-error
+achieved by the algorithm by calling the
+:func:`~NumericalInversePolynomial.u_error` method:
+
+    >>> from scipy.special import ndtr
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...         return np.exp(-0.5 * x*x)
+    ...     def cdf(self, x):
+    ...         return ndtr(x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInversePolynomial(dist, random_state=urng)
+    >>> rng.u_error(sample_size=100_000)
+    UError(max_error=8.785949745515609e-11, mean_absolute_error=2.9307548109436816e-11)
+
+:func:`~NumericalInversePolynomial.u_error` runs a monte carlo simulation with
+a given number of samples to estimate the u-error. In the above example,
+100,000 samples are used by the simulation to approximate the u-error. It
+returns the maximum u-error (``max_error``) and the mean absolute u-error
+(``mean_absolute_error``) in a ``UError`` namedtuple. As we can see,
+``max_error`` is below the default ``u_resolution`` (``1e-10``).
+
+It is also possible to evaluate the PPF of the given distribution once the
+generator is initialized:
+
+    >>> rng.ppf(0.975)
+    1.959963985701268
+    >>> norm.ppf(0.975)
+    1.959963984540054
+
+We can use this, for example, to check the maximum and mean absolute u-error:
+
+    >>> u = np.linspace(0.001, 0.999, num=1_000_000)
+    >>> u_errors = np.abs(u - dist.cdf(rng.ppf(u)))
+    >>> u_errors.max()
+    8.78600525666684e-11
+    >>> u_errors.mean()
+    2.9321444940323206e-11
+
+The approximate PPF method provided by the generator is much faster to
+evaluate than the exact PPF of the distribution.
+
+During setup, the CDF of the distribution is computed numerically but it is
+discarded once the setup is complete. To keep the approximated CDF, pass
+``keep_cdf=True`` during initialization:
+
+    >>> rng = NumericalInversePolynomial(dist, keep_cdf=True, random_state=urng)
+
+Now, the CDF can be evaluated by calling the
+:func:`~NumericalInversePolynomial.cdf` method:
+
+    >>> rng.cdf(1.959963984540054)
+    0.9750000000042454
+    >>> norm.cdf(1.959963984540054)
+    0.975
+
+We can use this to check if the integration error while computing the CDF
+exceeds :math:`0.05I_{0}\epsilon_{u}`. Here :math:`I_0` is
+:math:`\sqrt{2\pi}` (the normalization constant for the standard normal
+distribution):
+
+    >>> x = np.linspace(-10, 10, num=100_000)
+    >>> x_error = np.abs(dist.cdf(x) - rng.cdf(x))
+    >>> x_error.max()
+    4.506062190046123e-12
+    >>> I0 = np.sqrt(2*np.pi)
+    >>> max_integration_error = 0.05 * I0 * 1e-10
+    >>> x_error.max() <= max_integration_error
+    True
+
+Note that each CDF evaluation uses a Gauss-Lobatto quadrature which calls
```

Comment:
I would still mention that the CDF values computed in the setup are used and that only some further fine-tuning is needed, i.e. no need to do a full integration over the PDF

Reply:
> I would still mention that the CDF values computed in the setup are used and that only some further fine-tuning is needed, i.e. no need to do a full integration over the PDF

Sure, will mention this!

Reply:
Done.